src/catalog/function_catalog.cpp

```diff
@@ -0,0 +1,355 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// query_metrics_catalog.cpp
+//
+// Identification: src/catalog/query_metrics_catalog.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "catalog/function_catalog.h"
+#include "catalog/catalog_defaults.h"
+#include "catalog/catalog.h"
+#include "common/macros.h"
+#include "common/logger.h"
+#include <boost/algorithm/string/classification.hpp>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/foreach.hpp>
+
+namespace peloton {
+  namespace catalog {
+
+    FunctionCatalog *FunctionCatalog::GetInstance(
+        storage::Database *pg_catalog, type::AbstractPool *pool,
+        concurrency::Transaction *txn) {
+      static std::unique_ptr<FunctionCatalog> function_catalog(
+          new FunctionCatalog(pg_catalog, pool,txn));
+
+      return function_catalog.get();
+    }
+
+    FunctionCatalog::FunctionCatalog(
+        storage::Database *pg_catalog,
+        UNUSED_ATTRIBUTE type::AbstractPool *pool,
+        UNUSED_ATTRIBUTE concurrency::Transaction *txn)
+      : AbstractCatalog(FUNCTION_CATALOG_OID, FUNCTION_CATALOG_NAME,
+          InitializeSchema().release(), pg_catalog) {
+     
+
+      } 
+
+    FunctionCatalog::~FunctionCatalog() {}
+
+    /*@brief   private function for initialize schema of pg_database
+     * @return  unqiue pointer to schema
+     */
+    std::unique_ptr<catalog::Schema> FunctionCatalog::InitializeSchema() {
```

Comment:
I think in general this function could use a more detailed header and/or more comments to explain what all these seemingly arbitrary columns in the FunctionCatalog's schema are for.